packages/driver/src/cy/snapshots.ts

```diff
@@ -229,6 +229,11 @@ export const create = ($$: $Cy['$$'], state: StateFunc) => {
 
   const createSnapshot = (name, $elToHighlight, preprocessedSnapshot) => {
     Cypress.action('cy:snapshot', name)
+    // only take a snapshot if the AUT document is in state to prevent cypress-in-cypress like snapshots
+    // or if the snapshot has been taken in a spec bridge and has already been preprocessed
+    if (!preprocessedSnapshot && !state('document')) {
+      return null
+    }
```

Comment:
Does this fix just mean there are more missing screenshots? Can you share a few scenarios this is currently broken? I'd like to test the before & after locally. Does the comment on line 249 no long valid? This is where we are falling back to `window.document`....would it be more appropriate to query for the AUT iframe?

Reply:
> Does the comment on line 249 no long valid? This is where we are falling back to window.document

That's a good question. Seems like [these lines](https://github.com/cypress-io/cypress/blob/7c5ab8a9c4cdf58a5d3fa02870d72c2206cb9036/packages/driver/src/cy/snapshots.ts#L253-L254) should be updated/removed.

> would it be more appropriate to query for the AUT iframe?

I think if there's no `state('document')`, it won't be possible because that means we don't have synchronous access to the AUT.

Reply:
This actually brings up a good point, but I think it needs to stay. Main reason is logs streaming in from the secondary need to be cloned off a document, which in this case the primary document state is unloaded. But the backup isn't the AUT document in this case, rather the actual top document, which is OK since we only using the window to clone the node. If anything, we should probably use `window.document` here instead of `state("document")`

I'll play around with this to make sure it works as expect and will add a comment to explain this logic

Reply:
> Does this fix just mean there are more missing screenshots? Can you share a few scenarios this is currently broken? I'd like to test the before & after locally.

@emilyrohrbough you can run the snapshot tests from the driver locally, but the impact isn't exactly super noticeable there. Do you have `cypress-origin-providers` set up at all?

Reply:
Added comments in 228cef0524. I kept the current logic of using loaded document in state first before the specbridge/root document because of how many edge cases exist in snapshot code that could potentially break, but not be caught by this change. Here be 🐉 s.

Reply:
I can easily setup ` cypress-origin-providers` but even if I run the driver tests, what would I be looking for to verify this was acutally fixed? xhr requests?

Reply:
If you run the `snapshots.cy.ts` inside the `origin` driver tests, hover over the `(page load)` or `(new url)` logs to see no snapshot taken, where as before it would look something like this:
![Screen Shot 2022-11-04 at 1 07 54 PM](https://user-images.githubusercontent.com/3980464/200034819-c03fbb7d-9c8c-4c8a-b15f-1bd973157109.png)